clients/java/src/main/java/io/camunda/zeebe/client/ZeebeClient.java

```diff
@@ -618,6 +621,78 @@ static ZeebeClientCloudBuilderStep1 newCloudClientBuilder() {
    */
   UpdateJobCommandStep1 newUpdateJobCommand(ActivatedJob job);
 
+  /**
+   * Command to pin the Zeebe engine's internal clock to a specific time.
+   *
+   * <p>This method allows you to set a precise, static time for the clock using an epoch timestamp
+   * in milliseconds. Once pinned, the clock will remain at the specified time and will not advance
+   * until another <code>pin</code> or <code>reset</code> command is issued.
+   *
+   * <p>This command is useful for scenarios where you need to simulate process execution at a
+   * specific point in time.
+   *
+   * <pre>{@code
+   * final long pinnedTime = 1742461285000L; // Thu, Mar 20, 2025 09:01:25 GMT+0000
+   * zeebeClient
+   *  .newClockPinCommand(pinnedTime)
+   *  .send();
+   * }</pre>
+   *
+   * <p>The command is marked as <strong>experimental</strong> and may undergo changes or
+   * improvements in future releases.
+   *
+   * @param timestamp the epoch time in milliseconds to which the clock should be pinned
+   * @return a builder for the command
+   */
+  @ExperimentalApi("https://github.com/camunda/camunda/issues/21647")
+  ClockPinCommandStep1 newClockPinCommand(long timestamp);
```

Comment:
❌  I'm not sure about this approach, providing the `timestamp` or the `instant` at this step, usually at this step of the command chain we provide a `key` to identify the object to update/delete and then we use the second step to provide the value.

The better approach here is to do something like:
```
client.newClockPinCommand().pin(timestamp/instant).send()
```
This is aligned to all the others command that we have.

💭 Another approach is to have a single command:
```
client.newClockCommand().pin(timestamp/instant).send()
client.newClockCommand().reset(timestamp/instant).send()
```
But this is not fully aligned with the others command